### PR TITLE
[NCL-4992] create SCMRepositoryEndpoint integration test and add miss…

### PR DESCRIPTION
**Pull request by michalovjan**
…ing validation

### Checklist:

* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/SCMRepositoryEndpointTest.java [hunk 1]
```diff
@@ -59,4 +84,133 @@ public void testGetBuildConfigurationForARepository() throws ClientException {
...
+                .build();
+        //When
+        RepositoryCreationResponse response = repositoryClient.createNew(request);
+
+        //Then
+        assertThat(response).isNotNull();
+        assertThat(response.getRepository().getId()).isNotNull();
+
+        SCMRepository refreshed = repositoryClient.getSpecific(response.getRepository().getId());
+        assertThat(repository).isEqualToIgnoringGivenFields(refreshed, "id");
```
Comment (janinko) [inline integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/SCMRepositoryEndpointTest.java]: You have switched expected and actual

File: facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigurationProviderImpl.java [hunk 2]
```diff
@@ -219,7 +224,9 @@ public BuildConfigurationRevision createRevision(String id, BuildConfiguration b
                                                                             String sortingRsql,
                                                                             String query,
                                                                             String productVersionId) {
-
+        ValidationBuilder
+                .validateObject(null)
```
[the remark this answers is not in the document]
Comment (michalovjan) [reply facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigurationProviderImpl.java]: The reason was that I needed only to validate with method validateAgainstRepository(). I didn't have an Object to validate (only id), and the validated id didn't exactly belong to any of the validation groups (WhenUpdating, WhenDeleting...). The validateAgainstRepository is practically static, but as it returns "this", it is not. I've had to instantiate it the ValidationBuilder class and the most not confusing way was through validateObject method that has no groups.

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/SCMRepositoryEndpointTest.java [hunk 3]
```diff
@@ -59,4 +84,133 @@ public void testGetBuildConfigurationForARepository() throws ClientException {
...
+    public void shouldNotAllowUpdatingInternalUrl() throws ClientException {
+        //then
+        final String validInternalUrl = "ssh://github.com:22/pro12165cl/pnc2.git";
+        SCMRepository scmRepository = repositoryClient.getAll(null,null).iterator().next();
+        SCMRepository toUpdate = scmRepository.toBuilder().internalUrl(validInternalUrl).build();
+
+        //when/then
+        repositoryClient.update(toUpdate.getId(), toUpdate);
+
+        logger.debug("HIIIIIIIIIIIIIIIIIIIIIIIIIIII" + repositoryClient.getSpecific(toUpdate.getId()).toString());
```
[the remark this answers is not in the document]
Comment (michalovjan) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/SCMRepositoryEndpointTest.java]: Oh damn... That test was supposed to throw an exception when calling update(). Gotta get back to this one. Thx 4 the catch.

File: integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/SCMRepositoryEndpointTest.java (hunk 1 above)
Comment (michalovjan) [reply integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/SCMRepositoryEndpointTest.java]: Oh... my bad :D

File: facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java [hunk 4]
```diff
@@ -155,6 +155,18 @@ public RepositoryCreationResponse createSCMRepository(String scmUrl, Boolean pre
...
     }
 
+    @Override
+    public SCMRepository update(String id, SCMRepository restEntity) {
+        validateBeforeUpdating(id, restEntity);
+        SCMRepository before = getSpecific(id);
+        if (!before.getInternalUrl().equals(restEntity.getInternalUrl())) {
+            throw new InvalidEntityException("Updating internal URL is prohibited. SCMRepo: " + id);
+        }
+        log.debug("Updating entity: " + restEntity.toString());
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java]: Why do you use explicit `toString()`?

**Pull request merged**